sdk/confluent/Azure.ResourceManager.Confluent/tests/Scenario/ConfluentTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Confluent.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Confluent.Tests.Scenario
+{
+    public class ConfluentTests : ConfluentManagementTestBase
+    {
+        public ConfluentTests(bool async)
+           : base(async)
+        {
+        }
+
+        [OneTimeTearDown]
+        public void Cleanup()
+        {
+            CleanupResourceGroups();
+        }
+        [RecordedTest]
+        [TestCase]
+        public async Task CreateorUpdateConfluentOrg()
+        {
+            var confluentOrgCollection = GetConfluentOrganizationCollectionAsync();
+            ConfluentOfferDetail offerDetail = new ConfluentOfferDetail("confluentinc", "confluent-cloud-azure-stag", "confluent-cloud-azure-payg-stag", "Confluent Cloud - Pay as you Go", "P1M");
+            offerDetail.TermId = "gmz7xq9ge3py";
+            ConfluentUserDetail userDetail = new ConfluentUserDetail()
+            {
+                FirstName = "LiftrConfluent",
+                LastName = "User",
+                EmailAddress = $"liftrcftsdk{DateTime.Now:ddMMyyyy}@outlook.com",
+            };
+            ConfluentOrganizationData inputData = new ConfluentOrganizationData(DefaultLocation, offerDetail, userDetail);
+            var confluentNewOrg = await confluentOrgCollection.CreateOrUpdateAsync(WaitUntil.Completed, $"liftrcftsdk{DateTime.Now:ddMMyyyy}", inputData);
+            Assert.NotNull(confluentNewOrg);
+            Assert.NotNull(confluentNewOrg.Value.Id);
+            Assert.NotNull(confluentNewOrg.Value.Data.SsoUri);
+        }
+
+        [RecordedTest]
+        [TestCase]
+        public async Task ListConfluentOrgsInSubscription()
+        {
+            int count = 0;
+            await foreach (ConfluentOrganizationResource confluentOrganization in DefaultSubscription.GetConfluentOrganizationsAsync())
+            {
+                Console.WriteLine("Confluent Org: " + confluentOrganization.Id.Name);
+                ++count;
+            }
+            Assert.GreaterOrEqual(count, 1);
+        }
+
+        [RecordedTest]
+        [TestCase]
+        public async Task GetEnvironmentsInOrg()
+        {
+            await foreach (ConfluentOrganizationResource confluentOrganization in DefaultSubscription.GetConfluentOrganizationsAsync())
```

Comment:
All test cases should be re-runnable, no assumption of any pre-created resources existing.

Reply:
Updated